src/service/url-replacements-impl.js

```diff
@@ -14,510 +14,39 @@
  * limitations under the License.
  */
 
-import {accessServiceForOrNull} from '../access-service';
-import {cidFor} from '../cid';
-import {variantForOrNull} from '../variant-service';
-import {shareTrackingForOrNull} from '../share-tracking-service';
-import {dev, user, rethrowAsync} from '../log';
+import {dev, user} from '../log';
 import {documentInfoForDoc} from '../document-info';
 import {getServiceForDoc, installServiceInEmbedScope} from '../service';
-import {parseUrl, removeFragment, parseQueryString} from '../url';
-import {viewerForDoc} from '../viewer';
-import {viewportForDoc} from '../viewport';
-import {userNotificationManagerFor} from '../user-notification';
-import {activityFor} from '../activity';
+import {parseUrl} from '../url';
 import {isExperimentOn} from '../experiments';
-import {getTrackImpressionPromise} from '../impression.js';
 import {
-  VariableSource,
-  AsyncResolverDef,
-  ResolverReturnDef,
-  SyncResolverDef,
-  getNavigationData,
-  getTimingDataSync,
-  getTimingDataAsync,
+  installVarSubstitutionServiceForDoc,
+  installVarSubstitutionForEmbed,
+} from './var-substitution-impl';
+import {varSubstitutionForDoc} from '../var-substitution';
+import {
+    VariableSource,
```

Comment:
The `gulp check-types` throw errors because it doesn't know the Defs and the `VariableSource` type. I've noticed that we do add them to `externs` sometimes, should I do that instead? 

This still breaks linting for example giving me `VariableSource` is defined but never used problem. So not sure how to best fix this.